### PR TITLE
Implement connected components algorithm for graphs

### PR DESCRIPTION
**Pull request by nikalosa**
### **Describe your change:**
Find out connected components for given graph. 

* Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* This pull request is all my own work -- I have not plagiarized.
* I know that pull requests will not be merged if they fail the automated tests.
* This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* All new Python files are placed inside an existing directory.
* All filenames are in all lowercase characters with no spaces or dashes.
* All functions and variable names follow Python naming conventions.
* All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: graphs/connected_components.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    0: [1, 2, 3],
+    1: [0, 3],
+    2: [0],
+    3: [0, 1],
+    4: [],
+    5: [],
+}
+
+
+def dfs(graph, vert, visited):
```
Comment (mateuszz0000) [inline graphs/connected_components.py:28]: Please add typehints and doctests

Comment (nikalosa) [reply graphs/connected_components.py]: Done!

File: graphs/connected_components.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
+"""
+https://en.wikipedia.org/wiki/Component_(graph_theory)
+
+Finding connected components in graph
+
+"""
+
+test_graph_1 = {
```
Comment (mateuszz0000) [inline graphs/connected_components.py:8]: @cclauss The same question like in #2114 review.

File: graphs/connected_components.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    [0, 1, 3, 2]
+    """
+
+    visited[vert] = True
+
+    connected_verts = []
+
+    # iterate over neighbours and
+    # call the same function for not-visited vertexes
+    for neighbour in graph[vert]:
```
Comment (mateuszz0000) [inline graphs/connected_components.py:44]: Please remove unecessary comments. Code should be readable by itself.

File: graphs/connected_components.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    This function takes graph as a parameter
+    and then returns the list of connected components
+    >>> connected_components(test_graph_1)
+    [[0, 1, 3, 2], [4, 5, 6]]
+    >>> connected_components(test_graph_2)
+    [[0, 1, 3, 2], [4], [5]]
+    """
+
+    n = len(graph)
+
```
Comment (mateuszz0000) [inline graphs/connected_components.py:62]: `n` --> `graph_size`

File: graphs/connected_components.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    # for each unused vertexes
+    # we have to call dfs function
+    # and get its components
+    for i in range(n):
+        if not visited[i]:
+            i_connected = dfs(graph, i, visited)
+            # when we get components append them to answer list
+            components_list.append(i_connected)
+
```
Comment (mateuszz0000) [inline graphs/connected_components.py:75]: Please remove unecessary comments. Code should be readable by itself.

File: graphs/connected_components.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    return components_list
+
+
+if __name__ == "__main__":
+    import doctest
+
+    doctest.testmod()
```
Comment (mateuszz0000) [inline graphs/connected_components.py:82]: Please provide simple example instead import doctests.

Comment (nikalosa) [reply graphs/connected_components.py:73]: Can you clarify this request? Do you mean deleting doctests and just add same test in the main?

**Pull request merged**